internal/kafka/internal/services/kafka.go

```diff
@@ -1376,3 +1421,21 @@ func (k *kafkaService) IsQuotaEntitlementActive(kafkaRequest *dbapi.KafkaRequest
 
 	return quotaService.IsQuotaEntitlementActive(kafkaRequest)
 }
+
+type ClusterSizeCount struct {
+	SizeId string
+	Count  int32
+}
+
+func (k *kafkaService) GetKafkaSizeCountOfEnterpriseCluster(clusterID string) ([]*ClusterSizeCount, error) {
```

Comment:
I think we could enhance this method to only return the streaming unit count per instance type. 
```suggestion
func (k *kafkaService) ComputeUsedStreamingUnitCountOfCluster(clusterID string) (StreamingUnitCountPerInstanceType, error) {
```

and StreamingUnitCountPerInstanceType  

```
type StreamingUnitCountPerInstanceType map[KafkaInstanceType]int
```

It also makes sense for this method to be moved onto cluster_service.go file.